packages/waas/src/payloads/signers/secp256r1.ts

```diff
@@ -0,0 +1,61 @@
+import { ethers } from "ethers";
+import { PayloadSigner } from "./index";
+
+export async function newSECP256R1Signer(privateKey: string): Promise<PayloadSigner> {
+  const [pubKey, privKey] = await loadSECP256R1Keys(privateKey)
+
+  const exPrivateKey = await window.crypto.subtle.exportKey('jwk', privKey)
+  const exPublicKey = await window.crypto.subtle.exportKey('jwk', pubKey)
+  const keys = [exPublicKey, exPrivateKey]
+
+  const encoder = new TextEncoder()
+  return {
+    privateKey: ethers.utils.hexlify(encoder.encode(JSON.stringify(keys))),
+    verifier: async () => {
+      const pubRaw = await window.crypto.subtle.exportKey('raw', pubKey)
+      return 'r1:'+ethers.utils.hexlify(new Uint8Array(pubRaw))
+    },
+    signMessage: async (message: string | Uint8Array) => {
+      if (typeof message === 'string') {
+        if (message.startsWith('0x')) {
+          message = message.slice(2)
+          message = ethers.utils.arrayify(message)
+        } else {
+          message = encoder.encode(message)
+        }
+      }
+      const signatureBuff = await window.crypto.subtle.sign({name: 'ECDSA', hash: {name: 'SHA-256'}}, privKey, message)
+      return 'r1:'+ethers.utils.hexlify(new Uint8Array(signatureBuff))
+    }
+  }
+}
+
+export async function newSECP256R1RandomSigner(): Promise<PayloadSigner> {
+  const generatedKeys = await window.crypto.subtle.generateKey(
+    {
+      name: "ECDSA",
+      namedCurve: "P-256",
+    },
+    true,
```

Comment:
should be false... `extractable` argument.. so the private key is not extractable.

FYI, the browser will persist the private key for you between sessions, etc. so it will be there. Just need to check the APIs etc. I believe it stored in IndexedDB. If you're not sure, you can ping Corban and ask him, he's done some window.crypto stuff before.

Reply:
I couldn't find a way to persist the non-exportable key. Thank you for pointing me to this solution. I already have found a couple of examples.

Reply:
Cool.. were you able to find a way to persist the non-exportable key..?

because, tbh, if we can't use the non-exportable key, then there is no point in adding R1 support at all, and we should just skip it. The only value to r1 is we can have non-exportable private keys.

Reply:
This is what I have found:
https://gist.github.com/saulshanabrook/b74984677bccd08b028b30d9968623f5

They claim that you can store it in IndexedDB as non-exportable object. However they are not sure if it's more secure.